third_party/terraform/data_sources/data_source_google_sql_ca_certs.go

```diff
@@ -0,0 +1,108 @@
+package google
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceGoogleSQLCaCerts() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGoogleSQLCaCertsRead,
+
+		Schema: map[string]*schema.Schema{
+			"active_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"certs": {
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cert": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"common_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"create_time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"expiration_time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"sha1_fingerprint": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+				Computed: true,
+			},
+			"instance": {
```

Comment:
Instead of separate `instance` and `instance_self_link`, let's have just one field that can take the instance name or self link. You can do this by using the `compareSelfLinkOrResourceName` `DiffSuppressFunc`- the logic you already have for the project field value will take care of getting the correct project.

Reply:
This is great as it simplifies the implementation. Done.